tests/test_script_helper.py

```diff
@@ -1,222 +1,284 @@
+    plt.plot(time, np_recording_vec_reduced,
+             label='new reduction - voltage at soma (mV)', c='g')
+    plt.title("differences in Reduction - %s Hz (sum = %s)" %
+              (reduction_frequency, rmsd_two_reduced_vecs))
+    plt.legend(loc='lower right')
+    plt.show()
+
+
+def create_model(morphology_file, model_obj_name, instance_name, create_type):
+    '''Creates a full morphology cell instance from Neurolucida morphology model template filenames
+
+    This function is called to create the original instance to be reduced and
+    to create the control cell
+
+    TODO: To support non-Neurolucida morphology files, this function should be CHANGED.
     '''
-    if purpose == "reduction":
-        instance_name = "original_cell"
-    else:
-        instance_name = "control_cell"
-    
-    if create_type in ['basic' , 'human']:
-        # creates instance
-        h(instance_name + " = new " + model_obj_name + "()")
-        
+    assert instance_name in ('original_cell', 'control_cell'), \
+        "name must be one of ('original_cell', 'control_cell')"
+
+    h("{objref %s}" % instance_name)
+    model = dict(instance_name=instance_name, model_obj_name=model_obj_name)
+    if create_type in ('basic', 'human'):
+        h("{instance_name} = new {model_obj_name}()".format(**model))
+
         # instantiates according to morphology using import3d
         nl = h.Import3d_Neurolucida3()
         nl.quiet = 1
         nl.input(morphology_file)
-        imprt = h.Import3d_GUI(nl, 0)
+        import_3d = h.Import3d_GUI(nl, 0)
 
-        
-        instance_created = h.original_cell if purpose == "reduction" else h.control_cell
+        instance_created = getattr(h, instance_name)
         # associates cell instance with the morphology file
-        imprt.instantiate(instance_created)    
-        instance_created.complete_full_model_creation() # this function should be included in the model.hoc file given as a parameter
-        
-    elif create_type in ['bbp', 'bbpactive']:
-        h(instance_name + " = new " + model_obj_name + '(1,"' + morphology_file +'")')
-        h(instance_name + '=' + instance_name + '.CellRef')
-        instance_created = h.original_cell if purpose == "reduction" else h.control_cell
-    elif create_type == 'bbpnew':
-        cwd = os.getcwd()
-        os.chdir(morphology_file[:morphology_file.rindex('/')])
-        h(instance_name + " = new " + model_obj_name + '(0)')
-        os.chdir(cwd)
-        instance_created = h.original_cell if purpose == "reduction" else h.control_cell
+        import_3d.instantiate(instance_created)
+        # this function should be included in the model.hoc file given as a parameter
+        instance_created.complete_full_model_creation()
+    elif create_type in ('bbp', 'bbpactive'):
+        h('{instance_name} = new {model_obj_name}(1, "{morphology_file}")'.format(
+            morphology_file=morphology_file, **model))
```

Comment:
I see that it is cleaner, but isn't its harder to understand in this way?

Reply:
I'm biased, b/c I made the change, but I would argue that this reads more like hoc, because you don't have to parse the `+` signs.  It also makes it more clear that there is a set of `"` around the morphology file, for instance.